functions/Get-DbaHelpIndex.ps1

```diff
@@ -0,0 +1,596 @@
+.LINK
+https://dbatools.io/Get-DbaHelpIndex
+
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB
+	Returns information on all indexes on the MyDB database on the localhost.
+
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB,MyDB2
+	Returns information on all indexes on the MyDB, MyDB2 databases
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB -ObjectName dbo.Table1
+	Returns index information on the object dbo.Table1 in the database MyDB
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Databases MyDB -ObjectName dbo.Table1 -IncludeStats
+	Returns information on the indexes and statistics for the table dbo.Table1 in the MyDB database
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Database MyDB -ObjectName dbo.Table1 -IncludeDataTypes
+	Returns the index information for the table dbo.Table1 in the MyDB database, and includes the data types for the key and include columns
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Database MyDB -ObjectName dbo.Table1 -FormatResults
+	Returns the index information for the table dbo.Table1 in the MyDB database, and returns the numerical data with separators to make it more readable (ie 1234 becomes 1,234)
+	
+.EXAMPLE
+Get-DbaHelpIndex -SqlServer localhost -Database MyDB -IncludeStats -FormatResults
+	Returns the index information for all index in the MyDB database, as well as statistics, and formats the numerical data to be more redable	
+	
+#>
+	
+	[CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[object[]]$SqlServer,
+		[object]$SqlCredential,
+		[string]$ObjectName,
+		[switch]$IncludeStats,
+		[switch]$IncludeDataTypes,
+		[switch]$FormatResults
+	)
+	DynamicParam { if ($SqlServer) { return Get-ParamSqlDatabases -SqlServer $SqlServer -SqlCredential $SqlCredential } }
+	
+	BEGIN
+	{
+		
+		#Add the table predicate to the query
+		IF (!$ObjectName)
+		{
+			$TablePredicate = " @TableName NVARCHAR(256)";
+		}
+		ELSE
+		{
+			$TablePredicate = " @TableName NVARCHAR(256) = '$ObjectName'";
+		}
+		
+		
+		#Figure out if we are including stats in the results
+		IF ($IncludeStats)
+		{
+			$IncludeStatsPredicate = "";
+		}
+		ELSE
+		{
+			$IncludeStatsPredicate = "WHERE IndexType != 'STATISTICS'";
+		}
+		
+		#Data types being returns with the results?
+		IF ($IncludeDataTypes)
+		{
+			$IncludeDataTypesPredicate = '@IncludeDataTypes BIT = 1';
+		}
+		ELSE
+		{
+			$IncludeDataTypesPredicate = '@IncludeDataTypes BIT = 0';
+		}
+		
+#region SizesQuery		
+		$SizesQuery = @"
+SET NOCOUNT ON;
+SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED;
+	DECLARE
+		$TablePredicate, 
+		$IncludeDataTypesPredicate
+		;
+
+DECLARE @IndexUsageStats TABLE
+    (
+      object_id INT ,
+      index_id INT ,
+      user_scans BIGINT ,
+      user_seeks BIGINT ,
+      user_updates BIGINT ,
+      user_lookups BIGINT ,
+      last_user_lookup DATETIME2(0) ,
```

Comment:
The datetime2 is not the only issue. Also some columns on the DMV only appear later. For example the filter (not mentioned here, should be nice if you can update with it :-) ), and compression.
We have other command (https://github.com/sqlcollaborative/dbatools/blob/master/functions/Find-SqlDuplicateIndex.ps1) that use those and I had to use 2 different sql queries in order to support both.